app/src/main/java/com/example/myapp/composable/screen/HomeScreen.kt

```diff
@@ -9,80 +9,121 @@ import androidx.compose.foundation.layout.fillMaxSize
 import androidx.compose.foundation.layout.fillMaxWidth
 import androidx.compose.foundation.layout.padding
 import androidx.compose.foundation.layout.size
-import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.layout.width
 import androidx.compose.foundation.lazy.grid.GridCells
+import androidx.compose.foundation.lazy.grid.GridItemSpan
 import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
 import androidx.compose.foundation.lazy.grid.itemsIndexed
 import androidx.compose.foundation.lazy.grid.rememberLazyGridState
-import androidx.compose.foundation.lazy.itemsIndexed
 import androidx.compose.foundation.shape.RoundedCornerShape
 import androidx.compose.foundation.text.KeyboardActions
 import androidx.compose.foundation.text.KeyboardOptions
 import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.KeyboardArrowLeft
+import androidx.compose.material.icons.filled.KeyboardArrowRight
 import androidx.compose.material.icons.filled.Refresh
 import androidx.compose.material.icons.filled.Search
-import androidx.compose.material.icons.materialIcon
 import androidx.compose.material3.Button
-import androidx.compose.material3.ButtonDefaults
 import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
 import androidx.compose.material3.OutlinedTextField
 import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
 import androidx.compose.runtime.getValue
 import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.remember
+import androidx.compose.runtime.saveable.rememberSaveable
 import androidx.compose.runtime.setValue
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.graphics.Color
 import androidx.compose.ui.graphics.vector.ImageVector
 import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.KeyboardType
 import androidx.compose.ui.unit.dp
 import androidx.compose.ui.unit.sp
 import androidx.navigation.NavController
-import com.example.myapp.Retrofit.ArtistData
-import com.example.myapp.Retrofit.LiveData
 import com.example.myapp.composable.component.HomeItem
 import com.example.myapp.data.MyViewModel
-import java.util.Locale.filter
 
 
 @Composable
 fun HomeScreen(myViewModel: MyViewModel, navController: NavController) {
-    val isLoaded = myViewModel.isLoaded.value
+    val isLoaded = myViewModel.isArtistListLoaded.value
+
+    val size = 20
+
+    val lastPage = myViewModel.pageNum
+
+    val pageNum = rememberSaveable {
+        mutableStateOf(1)
+    }
+
+    val pageNumText = rememberSaveable {
+        mutableStateOf(pageNum.value.toString())
+    }
+
+    val artistName = rememberSaveable {
+        mutableStateOf("")
+    }
+
+    val artistNametext = rememberSaveable {
```

Comment:
```suggestion
    val artistNameText = rememberSaveable {
```

camel case는 단어 단위로 대문자로 씁니다.